### DISCUSSION TITLE
Fast cleanup?

### DISCUSSION THREAD

**Discussion by BondByteBlaster** [datalust/seq-tickets#2159] [Help]
Hi

Our SEQ installation warned of low disk space (2 GB), it went so far as to discard some logs. I found SEQ log files that took up nearly 50 GB. In an attempt to clean up, I tried running "manual retention" but only freed up additional 0,5 GB. I encountered multiple problems as it was unable to perform the retention jobs, mentioning something about indexing errors. It quickly became unclear which jobs it succeeded or failed to run.

Questions:

How does one quickly and safely delete log data to free up space?
How can one check which tasks have been executed and what the outcome was?
How does one search for log events based on disk size? Anre delete them?

Regards

**Comment by liammclennan** [comment]
Hi @BondByteBlaster 

The easiest solution is to make more disk space available without deleting anything from Seq. If that is not an option then the next best is to add a 'Delete all events' retention policy to delete your oldest events. 'Manually delete events' will not work well in this situation.

To answer your questions:

_How does one quickly and safely delete log data to free up space?_ 

See above.

_How can one check which tasks have been executed and what the outcome was?_

View the logs in `/Logs`. 

_How does one search for log events based on disk size? Anre delete them?_

Delete events using retention policies, or use API key filters to avoid storing them at all. To identify large events you can use the 'Space by Event Type' query in your queries list, which is:

```sql
select count(*) as Occurrences,
       sum(Length(ToJson(@Data))) as TotalChars,
       ToHexString(first(@EventType)) as EventType,
       first(@MessageTemplate) as MessageTemplate
from stream
group by @EventType
order by TotalChars desc
limit 10
```
